python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr_serverless.py

```diff
@@ -279,24 +283,31 @@ def _read_messages(self, context: OpExecutionContext, response: "GetJobRunRespon
             "stderr": sys.stderr,
         }
 
-        # TODO: do this in a background thread in real-time once https://github.com/dagster-io/dagster/pull/24098 is merged
-        for output_stream in output_streams:
-            output_file = output_files[output_stream]
-            context.log.debug(
-                f"[pipes] Reading AWS CloudWatch logs from group {log_group} stream {log_stream}/{output_stream}"
-            )
-            self.message_reader.consume_cloudwatch_logs(
-                log_group,
-                f"{log_stream}/{output_stream}",
-                start_time=int(
-                    response["jobRun"]
-                    .get("attemptCreatedAt", response["jobRun"]["createdAt"])
-                    .timestamp()
-                    * 1000
-                ),
-                output_file=output_file,
+        # update MessageReader params so it can start receiving messages
+        if isinstance(self.message_reader, PipesCloudWatchMessageReader):
+            self.message_reader.update_params(
+                {
+                    "log_group": log_group,
+                    "log_stream": f"{log_stream}/stdout",
+                }
             )
 
+        # now add LogReaders for stdout and stderr logs
+        if isinstance(self.message_reader, PipesThreadedMessageReader):
+            for output_stream in output_streams:
+                output_file = output_files[output_stream]
+                context.log.debug(
+                    f"[pipes] Adding PipesCloudWatchLogReader for group {log_group} stream {log_stream}/{output_stream}"
+                )
+                self.message_reader.add_log_reader(
```

Comment:
@schrockn here is an example of calling `.add_log_reader`. 

`log_stream` is only known after the external process has been launched so we have to manually add it here.